test/Annotation/AbstractBuilderTestCase.php

```diff
@@ -215,6 +221,28 @@ public function testAllowsComposingChildEntities(): void
         self::assertInstanceOf(InputFilterInterface::class, $composed);
         self::assertTrue($composed->has('username'));
         self::assertTrue($composed->has('password'));
+        $usernameInput = $composed->get('username');
+        self::assertInstanceOf(Input::class, $usernameInput);
+        $validatorChain = $usernameInput->getValidatorChain();
+        self::assertInstanceOf(ValidatorChain::class, $validatorChain);
+        $usernameValidators = $validatorChain->getValidators();
+        self::assertCount(2, $usernameValidators);
+        self::assertInstanceOf(NotEmpty::class, $usernameValidators[0]['instance']);
+        self::assertInstanceOf(StringLength::class, $usernameValidators[1]['instance']);
+        $usernameFilters = $usernameInput->getFilterChain()->getFilters()->toArray();
+        self::assertCount(1, $usernameFilters);
+        self::assertInstanceOf(StringTrim::class, $usernameFilters[0]);
+
+        $passwordInput = $composed->get('password');
+        self::assertInstanceOf(Input::class, $passwordInput);
+        $validatorChain     = $passwordInput->getValidatorChain();
+        $passwordValidators = $validatorChain->getValidators();
+        self::assertCount(1, $passwordValidators);
+        self::assertInstanceOf(EmailAddress::class, $passwordValidators[0]['instance']);
+
+        $passwordFilters = $passwordInput->getFilterChain()->getFilters()->toArray();
+        self::assertCount(1, $passwordFilters);
+        self::assertInstanceOf(StringTrim::class, $passwordFilters[0]);
```

Comment:
Can we split it out to a smaller test highlighting just what is being fixed? :thinking:

Reply:
Without the fix, `username` and `password` result indeed with no validator at all, wrongfully.